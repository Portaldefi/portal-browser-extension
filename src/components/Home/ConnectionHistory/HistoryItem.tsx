import React from 'react';
import { Grid, Header } from 'semantic-ui-react';

interface IConnectionItem {
  url: string,
  timestamp: string
};

export default ({ url, timestamp }: IConnectionItem) => {
  return (
    <Grid className='p-16 py-9 bg-white'>
      <Grid.Column width='sixteen' textAlign='left'>
        <Header as='p' className='description'>{url}</Header>
        <Header as='p' className='comment'>{timestamp}</Header>
      </Grid.Column>
    </Grid>
  )
}