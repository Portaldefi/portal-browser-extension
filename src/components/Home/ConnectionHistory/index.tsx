import React, { useCallback } from 'react';
import { useNavigate } from 'react-router-dom';
import { Button, Grid, Header, List } from 'semantic-ui-react';

import HistoryItem from './HistoryItem';

export default () => {
  const navigate = useNavigate();

  const handleClickConnection = useCallback(() => {
    navigate('/connection-detail')
  }, []);

  return (
    <Grid className='recent-connection'>
      <Grid.Row>
        <Grid.Column width='ten' textAlign='left'>
          <Header as='p' className='subtitle p-16 py-9'>Last Connections</Header>
        </Grid.Column>
        <Grid.Column width='six' textAlign='right'>
          <Button className='blank-button'>Clear</Button>
        </Grid.Column>
      </Grid.Row>
      <Grid.Row className=''>
        <List className='w-100'>
          <List.Item onClick={handleClickConnection}>
            <List.Content>
              <HistoryItem url='https://grove.chat/' timestamp='Tue, Apr. 26, 2022 5:00 am' />
            </List.Content>
          </List.Item>
          <List.Item onClick={handleClickConnection}>
            <List.Content>
              <HistoryItem url='https://grove.chat/' timestamp='Sat, Apr. 23, 2022 8:30 am' />
            </List.Content>
          </List.Item>
          <List.Item onClick={handleClickConnection}>
            <List.Content>
              <HistoryItem url='https://grove.chat/' timestamp='Fri, Apr. 22, 2022 12:00 pm' />
            </List.Content>
          </List.Item>
        </List>
      </Grid.Row>
    </Grid>
  )
}