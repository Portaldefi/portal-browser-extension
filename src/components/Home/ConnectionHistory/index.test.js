import ConnectionHistory from './index';
import { create, act } from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { List } from 'semantic-ui-react';

describe('ConnectionHistory', () => {
  it('should work', () => {
    let tree;
    act(() => {
      tree = create(
        <BrowserRouter>
          <ConnectionHistory />
        </BrowserRouter>
      );
    });

    expect(tree).toMatchSnapshot();
  });
  it('button click', () => {
    let tree;
    act(() => {
      tree = create(
        <BrowserRouter>
          <ConnectionHistory />
        </BrowserRouter>
      );
    });

    const items = tree.root.findAllByType(List.Item);
    act(() => items[0].props.onClick());
    expect(tree).toMatchSnapshot();
  });
  afterAll(() => jest.resetModules());
});