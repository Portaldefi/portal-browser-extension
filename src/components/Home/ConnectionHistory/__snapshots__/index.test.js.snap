// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectionHistory button click 1`] = `
<div
  className="ui grid recent-connection"
>
  <div
    className="row"
  >
    <div
      className="left aligned ten wide column"
    >
      <p
        className="ui header subtitle p-16 py-9"
      >
        Last Connections
      </p>
    </div>
    <div
      className="right aligned six wide column"
    >
      <button
        className="ui button blank-button"
        onClick={[Function]}
      >
        Clear
      </button>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="ui list w-100"
      role="list"
    >
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="content"
        >
          <div
            className="ui grid p-16 py-9 bg-white"
          >
            <div
              className="left aligned sixteen wide column"
            >
              <p
                className="ui header description"
              >
                https://grove.chat/
              </p>
              <p
                className="ui header comment"
              >
                Tue, Apr. 26, 2022 5:00 am
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="content"
        >
          <div
            className="ui grid p-16 py-9 bg-white"
          >
            <div
              className="left aligned sixteen wide column"
            >
              <p
                className="ui header description"
              >
                https://grove.chat/
              </p>
              <p
                className="ui header comment"
              >
                Sat, Apr. 23, 2022 8:30 am
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="content"
        >
          <div
            className="ui grid p-16 py-9 bg-white"
          >
            <div
              className="left aligned sixteen wide column"
            >
              <p
                className="ui header description"
              >
                https://grove.chat/
              </p>
              <p
                className="ui header comment"
              >
                Fri, Apr. 22, 2022 12:00 pm
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ConnectionHistory should work 1`] = `
<div
  className="ui grid recent-connection"
>
  <div
    className="row"
  >
    <div
      className="left aligned ten wide column"
    >
      <p
        className="ui header subtitle p-16 py-9"
      >
        Last Connections
      </p>
    </div>
    <div
      className="right aligned six wide column"
    >
      <button
        className="ui button blank-button"
        onClick={[Function]}
      >
        Clear
      </button>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="ui list w-100"
      role="list"
    >
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="content"
        >
          <div
            className="ui grid p-16 py-9 bg-white"
          >
            <div
              className="left aligned sixteen wide column"
            >
              <p
                className="ui header description"
              >
                https://grove.chat/
              </p>
              <p
                className="ui header comment"
              >
                Tue, Apr. 26, 2022 5:00 am
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="content"
        >
          <div
            className="ui grid p-16 py-9 bg-white"
          >
            <div
              className="left aligned sixteen wide column"
            >
              <p
                className="ui header description"
              >
                https://grove.chat/
              </p>
              <p
                className="ui header comment"
              >
                Sat, Apr. 23, 2022 8:30 am
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="content"
        >
          <div
            className="ui grid p-16 py-9 bg-white"
          >
            <div
              className="left aligned sixteen wide column"
            >
              <p
                className="ui header description"
              >
                https://grove.chat/
              </p>
              <p
                className="ui header comment"
              >
                Fri, Apr. 22, 2022 12:00 pm
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
